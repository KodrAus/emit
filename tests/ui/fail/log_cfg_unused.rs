#![feature(stmt_expr_attributes, proc_macro_hygiene)]
#![deny(unused_variables)]

fn main() {
    let a = String::from("hello");

    // Unused by the log statement
    let c = 42;

    emit::info!("A log with cfgs {#[cfg(disabled)] b: 17}",
        a,
        #[emit::as_debug]
        #[cfg(disabled)]
        c,
        d: String::from("short lived!"),
    );
}
