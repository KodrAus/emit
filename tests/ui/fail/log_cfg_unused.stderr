error: unused variable: `c`
 --> $DIR/log_cfg_unused.rs:8:9
  |
8 |     let c = 42;
  |         ^ help: if this is intentional, prefix it with an underscore: `_c`
  |
note: the lint level is defined here
 --> $DIR/log_cfg_unused.rs:2:9
  |
2 | #![deny(unused_variables)]
  |         ^^^^^^^^^^^^^^^^
